### PR TITLE
add GetEventWithStartTime

### PR DESCRIPTION
**Pull request by cenkore**
ref #325 

新增的一个func，你检查一下哈，谢谢。

### CODE REVIEW

File: replication/binlogstreamer.go [hunk 1]
```diff
@@ -36,6 +37,27 @@ func (s *BinlogStreamer) GetEvent(ctx context.Context) (*BinlogEvent, error) {
 	}
 }
 
+// Get the binlog event with starttime, if current binlog event timestamp smaller than specify starttime
+// return nil event 
+func (s *BinlogStreamer) GetEventWithStartTime(ctx context.Context,startTime time.Time) (*BinlogEvent, error) {
+	if s.err != nil {
+		return nil, ErrNeedSyncAgain
+	}
+	startTime,_ = time.ParseInLocation(mysql.TimeFormat,startTime.Format(mysql.TimeFormat),time.Local)
```
[the remark this answers is not in the document]
Comment (cenkore) [reply replication/binlogstreamer.go]: ah , yes . I will pull a new version later

File: replication/binlogstreamer.go [hunk 2]
```diff
@@ -36,6 +36,26 @@ func (s *BinlogStreamer) GetEvent(ctx context.Context) (*BinlogEvent, error) {
...
 
+// Get the binlog event with starttime, if current binlog event timestamp smaller than specify starttime
+// return nil event 
+func (s *BinlogStreamer) GetEventWithStartTime(ctx context.Context,startTime time.Time) (*BinlogEvent, error) {
+	if s.err != nil {
+		return nil, ErrNeedSyncAgain
+	}
+	startUnix := startTime.Unix()
+	select {
+	case c := <-s.ch:
```
Comment (siddontang) [inline replication/binlogstreamer.go]: is it better to loop check all the time until the Timestamp is larger than the startUnix, and then return the event?

but this depends on your case, if you think returning nil is fine, I will merge it.

Comment (cenkore) [reply replication/binlogstreamer.go]: Yes, it is suitable for my case . thanks.

**Comment by siddontang** [comment]
Thank you @cenkore

**Comment by xiaohuang1234** [comment]
how to get the startTime

**Comment by cenkore** [comment]
this pr is just used to some certain scenarios that you specify the startTime instead of binPos.

**Pull request merged**